nopassword/views.py

```diff
@@ -1,58 +1,48 @@
 # -*- coding: utf-8 -*-
-from django.conf import settings
-from django.contrib.auth import login as auth_login
-from django.contrib.auth import logout as auth_logout
-from django.contrib.auth import authenticate
-from django.contrib.auth.views import login as django_login
-from django.http import Http404
-from django.shortcuts import get_object_or_404, redirect, render
-
-from .forms import AuthenticationForm
-from .models import LoginCode
-from .utils import get_username, get_username_field
-
-
-def login(request):
-    if request.method == 'POST':
-        form = AuthenticationForm(data=request.POST)
-        if form.is_valid():
-            code = LoginCode.objects.filter(**{
-                'user__%s' % get_username_field(): request.POST.get('username')
-            })[0]
-            code.next = request.GET.get('next')
-            code.save()
-            code.send_login_code(
-                secure=request.is_secure(),
-                host=request.get_host(),
-            )
-            return render(request, 'registration/sent_mail.html')
-
-    return django_login(request, authentication_form=AuthenticationForm)
-
-
-def login_with_code(request, login_code):
-    code = get_object_or_404(LoginCode.objects.select_related('user'), code=login_code)
-    return login_with_code_and_username(request, username=get_username(code.user),
-                                        login_code=login_code)
-
-
-def login_with_code_and_username(request, username, login_code):
-    code = get_object_or_404(LoginCode, code=login_code)
-    login_with_post = getattr(settings, 'NOPASSWORD_POST_REDIRECT', True)
```

Comment:
Backward incompatible: `NOPASSWORD_POST_REDIRECT` not checked anymore.

Reply:
Why do we want to remove this configuration?

Reply:
I think using a custom template with js to make a separate `POST` request shouldn't be the default implementation.
It pretends to use `GET` correctly (no change in server state, idempotent), but eventually the `GET` request will do a state change (login).

In my opinion the proper way would be to render a template (without js) that has a form rendered which just says sth like `You can login by submitting this form`. The form will then do the `POST` request, but only if the USER does that.

If a vendor doesn't want a user to submit the login form, they can optionally active the login by a `GET` request.
The setting should be named then accordingly, like `NOPASSWORD_LOGIN_ON_GET`.

Note that this flow is also used for confirming an email address in allauth, see `ACCOUNT_CONFIRM_EMAIL_ON_GET` in https://django-allauth.readthedocs.io/en/latest/configuration.html

Should we change it like this?

Reply:
Sounds like a good idea to change the name to `NOPASSWORD_LOGIN_ON_GET`.